### ISSUE TITLE
Midpoint Review - Accessibility Feedback - Unified Appointment Experience, Appointments, Community Care Direct Scheduling

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#85828]
## Next Steps for the VFS team

- [x] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [ ] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

- Sorry for the slow feedback on this, and thank you for your patience!
- Overall, I really like the direction this is going! I'm going to give you some really nitpicky stuff here but I think you're on the right track.

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [x] **Should:** On the main [Appointments page](https://www.figma.com/design/DsRXEFiYLCFnY5nBkp9Dc4/CC-Referral-%7C-Appointments-FE?node-id=1899-75962&t=TAsRF6EWnpfibZSR-4), I think the card at the top of the page for the community care referral would benefit from a heading. It doesn't need to be a visible heading, but something to indicate that the card exists would be good. (I would also be 100% okay with a visible heading if you think the design can accommodate it!) 
   
   My thinking here is that a screen reader user who's used to coming to the Appointments page is probably going to navigate by heading most of the time, and will miss that this card has been added if there's not something added to the heading structure. 
   
   I would recommend including the expiration tag as part of the heading so that information doesn't get missed. Assuming the heading is hidden, you would end up with something like:
   ```
   <h2>
      <span class="whatever-the-tag-classes-are">Expires [date]</span>
      <span class="sr-only">Community Care Referral</span>
   </h2>
   ```
   That should provide enough structure for screen reader users without changing the visual presentation.
- [x] **Must:** The ["Your preferred provider" card](https://www.figma.com/design/DsRXEFiYLCFnY5nBkp9Dc4/CC-Referral-%7C-Appointments-FE?node-id=1899-75962&t=TAsRF6EWnpfibZSR-0) also needs a heading. My recommendation here is that "Your preferred provider" be an H2 equal to "All providers." I think if you can style it as a recognizable heading (again, matching "All providers") that's a net benefit for all users, but keeping it styled as ordinary text is fine so long as it's coded as a heading for screen reader users.
- [x] **Should:** The contrast on the ["Your preferred provider" card](https://www.figma.com/design/DsRXEFiYLCFnY5nBkp9Dc4/CC-Referral-%7C-Appointments-FE?node-id=1899-75962&t=TAsRF6EWnpfibZSR-0) for "Your preferred provider" against the gray background meets WCAG AA requirements, but it's pretty close. I don't remember if that text color against the gray background is already in use elsewhere, but I'd encourage you to try something darker. I don't think you'd lose much by just making it black.
- [x] **Engineering note:** Looking at the [list of providers available to schedule](https://www.figma.com/design/DsRXEFiYLCFnY5nBkp9Dc4/CC-Referral-%7C-Appointments-FE?node-id=1485-96748&t=TAsRF6EWnpfibZSR-4), I'm going to recommend coding this as a `ul` unordered list, with each available provider as a `li` list item. This might already be what you're intending, but I just want to call that out for its potential accessibility benefits. When coded as a list, screen reader users have a couple of extra tools for moving through that list efficiently.
- [x] **Must:** This might also already be what's intended, but note that each "Review available appointments" link will need an `aria-label` to give them unique accessible names for screen reader users. Something like `<a href="..." aria-label="Review available appointments for [provider name]">Review available appointments</a>`.
- [x] **Must:** For the [list of providers available to schedule](https://www.figma.com/design/DsRXEFiYLCFnY5nBkp9Dc4/CC-Referral-%7C-Appointments-FE?node-id=1485-96748&t=TAsRF6EWnpfibZSR-4), the "+5 more providers" needs to be a button rather than a link. There are some old instances of this being implemented with a link, but [design system guidance](https://design.va.gov/patterns/help-users-to/navigate-a-long-list) should reflect that it should be a button now. There are some content and accessibility requirements in that guidance for coding that element -- please reach out if anything isn't clear.
- [x] **Consider:** I feel like we talked about this in office hours and I don't remember what I recommended or if I told you to do the opposite of this... but I think it's worth moving the ["Sort providers by" dropdown](https://www.figma.com/design/DsRXEFiYLCFnY5nBkp9Dc4/CC-Referral-%7C-Appointments-FE?node-id=1485-96736&t=TAsRF6EWnpfibZSR-4) to the filters page, and updating the "Filters" link to "Filter and sort" or "Refine these results" or something like that.
   
   Here's where this is coming from: If I'm a screen reader user, my expectation is that after I sort by provider, my keyboard focus should move to the results, eg. moving to the summary text that says how many results are shown. That focus management notifies users that the sort action successfully occurred, and it puts you in the next logical place for what you're trying to do. But that means skipping over the link to the filters page, which might be missed by screen reader users. Likewise, after I come back from applying filters, my expectation would be to go straight to the results --- so if we flipped the order of the the sort and filter elements, you'd have the same issue. There's not a clear hierarchy --- sort then filter then results, filter then sort then results --- that would justify moving focus somewhere other than the results.
   
   I think putting sort and filter together on that separate page might solve this. Something to explore!
   - [ ] **Conditional must:** If you don't decide to move sort to the filter page, then...
      - Add a "Sort" button to trigger the sort action. This is coming from [WCAG 3.2.2](https://www.w3.org/WAI/WCAG22/Understanding/on-input.html), selecting something from a dropdown shouldn't initiate any changes of context.
      - Add a visually hidden `aria-live` region that announces when the sort action is completed, eg. "Results sorted by distance" when that option is selected. This isn't ideal, but it's better than a screen reader user sorting results and hearing nothing to confirm that something happened.
      - Don't do any focus management, just keep focus where it is when clicking "Sort."
- [x] **Must:** Echoing Erin again, the summary text for the results (currently "# Results") needs to provide more information. Ideally this would be something like "Showing # results, sorted by distance, filtered by PT Practice of Downtown LA." That gets very wordy (especially if you add more filter criteria), so a more condensed "# results, sorted by distance, with 1 filter applied" would work too.
   
   If you do any focus management when sorting/filtering, this is the line of text I would expect to receive focus. It introduces the results and gives a summary of what's displayed (including the sort/filter actions applied). It also helps manage some cognitive load, eg. a user gets confused because they thought something should be displayed differently, now there's summary text to help indicate where things might have gone wrong.

## Governance team actions
- [x] Format feedback as individual tasks (check boxes)
- [x] Assign this ticket to the VFS team member that opened the Slack request
- [x] Add the VFS team product label
- [x] Add the VFS team the feature label (if applicable)
- [x] Add the touchpoint labels
- [x] Add the practice area labels
- [x] Add the Collaboration Cycle initiative milestone

**Comment by benbrasso-agile6** [comment]
* Added task card hidden heading markup (might get tricky or need to change as this pattern expands to include other tasks outside of community care referrals)

> Must: The "Your preferred provider" card also needs a heading. My recommendation here is that "Your preferred provider" be an H2 equal to "All providers." I think if you can style it as a recognizable heading (again, matching "All providers") that's a net benefit for all users, but keeping it styled as ordinary text is fine so long as it's coded as a heading for screen reader users.

Done

> Engineering note

Added as annotation

> Aria label 

Added as annotation 

> Consider combining filter and sort

[Completed](https://www.figma.com/design/DsRXEFiYLCFnY5nBkp9Dc4/CC-Referral-%7C-Appointments-FE?node-id=2169-36405&t=4LNO4gXSPPPUrflA-4)

> # of results read out

[Updated](https://www.figma.com/design/DsRXEFiYLCFnY5nBkp9Dc4/CC-Referral-%7C-Appointments-FE?node-id=2424-42620&t=4LNO4gXSPPPUrflA-4)

**Comment by benbrasso-agile6** [comment]
@Cieramaddox two tasks left in this ticket for you to review

**Comment by Cieramaddox** [comment]
> Should: The contrast on the "Your preferred provider" card
Removed this from the card altogether (adding the an H2 above)

> "+5 more providers" needs to be a button rather than a link
[Changed to secondary button. ](https://www.figma.com/design/DsRXEFiYLCFnY5nBkp9Dc4/CC-Referral-%7C-Appointments-FE?node-id=1874-38076&t=bJtUY6Nvucctb1x6-1)
